pkg/storage/storage.go

```diff
@@ -28,6 +28,8 @@ type IPPool interface {
 // Store is the interface that wraps the basic IP Allocation methods on the underlying storage backend
 type Store interface {
 	GetIPPool(ctx context.Context, ipRange string) (IPPool, error)
+	IsAllocatedClusterWide(ctx context.Context, ip net.IP) (bool, error)
```

Comment:
I wonder if it would make more sense for a separate `Store` implementation that uses the cluster-scoped CRD (instead of expanding on the interface).

It seems like if you are using the cluster-scoped IPReservation resource you won't need the namespace-scoped resource?

Reply:
I think this is a great recommendation, I went ahead and changed up how we access the "cluster wide" resources, making a `ClusterStore`, so we follow a similar pattern as:

ipam to make -> Store
ipam to make -> ClusterStore